src/hooks/useColorManipulation.ts

```diff
@@ -32,10 +32,18 @@ export function useColorManipulation<T extends AnyColor>(
   // Trigger `onChange` callback only if an updated color is different from cached one;
   // save the new color to the ref to prevent unnecessary updates
   useEffect(() => {
-    let newColor;
+    const { a, ...hsv } = hsva;
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const { a: _, ...cacheHsv } = cache.current.hsva;
+
+    // When alpha channel is changed, use cached RGB values to prevent rounding errors
+    const newColor = equalColorObjects(hsv, cacheHsv)
+      ? Object.assign({}, cache.current.color, { a })
```

Comment:
Hi! Thanks for the contribution!

We should keep in mind that `cache.current.color` can be a string, so this code fixes only object-based color models and (as far as I see) breaks string one (like `rgba(200, 200, 200, 0.5)`). But I think we're somewhere close)

Reply:
Thanks for the reply!
As you mentioned, I confirmed that on the local demo page, an error occurs when the alpha value changed in the following component:

- `RgbaStringPicker`
- `HslaStringPicker`
- `HsvaStringPicker`

I'll try to find a better solution.

Reply:
One idea is to add a function to the properties of `colorModel` that updates only the alpha value:

```typescript
// HsvaStringColorPicker
const colorModel: ColorModel<string> = {
  defaultColor: "hsva(0, 0%, 0%, 1)",
  toHsva: hsvaStringToHsva,
  fromHsva: hsvaToHsvaString,
  equal: equalColorString,
  // Detects alpha values from strings and replaces
  updateAlpha: updateAlphaFromString,
};

// HsvStringColorPicker
const colorModel: ColorModel<string> = {
  defaultColor: "hsv(0, 0%, 0%)",
  toHsva: hsvStringToHsva,
  fromHsva: hsvaToHsvString,
  equal: equalColorString,
  // Do nothing
  updateAlpha: (value) => value,
};
```

Then... 

```typescript
const newColor = equalColorObjects(hsv, cacheHsv)
  // Regardless of the type of `cache.current.color`, only the alpha value should be properly updated
  ? colorModel.updateColorAlpha( cache.current.color, a )
  : colorModel.fromHsva(hsva);
```

Reply:
You know what? It's actually a good idea!
If we make it optional (not required in every color model), it won't total damage bundle size dramatically.

Reply:
Ok, then I will try using this approach 👍
This property should only be added to components that support alpha:

```typescript
const isSupportAlpha = 'updateAlpha' in colorModel;
const newColor = equalColorObjects(hsv, cacheHsv)
  ? isSupportAlpha ? colorModel.updateColorAlpha( cache.current.color, a ) : cache.current.color
  : colorModel.fromHsva(hsva);
```